index.js

```diff
@@ -86,7 +82,7 @@ module.exports = (dest, opts) => {
 			}
 		}
 
-		opts.hasChanged(this, cb, file, newPath);
+		opts.hasChanged(this, file, newPath).then(() => cb());
```

Comment:
You can now move the `fsOperationFailed` error handling here instead of catching it in both `compareLastModifiedTime` and `compareSha1Digest`.